sdk/python/kfp/gcp.py

```diff
@@ -53,6 +53,12 @@ def _use_gcp_secret(task):
                         value=secret_volume_mount_path + secret_file_path_in_volume,
                     )
                 )
+                .add_env_variable(
+                    k8s_client.V1EnvVar(
+                        name='CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE',
```

Comment:
What does the CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE do?

Reply:
It sets credentials for gcloud. GOOGLE_APPLICATION_CREDENTIALS works as application default credentials but not gcloud.

We need to get gcloud working to submit a CMLE job.

Reply:
Also added comments.